packages/snap/src/rpc/evm/common/EVMSigner.ts

```diff
@@ -1,32 +1,48 @@
-import {
-  BroadcastTransactionParamsType,
+import { addressNListToBIP32, bip32Like } from '@shapeshiftoss/hdwallet-core'
+import type {
   BroadcastTransactionResponseType,
   EVMChainIds,
   GetAddressParamsType,
   GetAddressResponseType,
-  SignerGetAddressType,
-  SignerSignMessageType,
-  SignerSignTransactionType,
-  SignerVerifyMessageType,
+  SendTransactionParamsType,
   SignMessageParamsType,
   SignMessageResponseType,
-  SignTransactionParamsType,
-  SignTransactionResponseType,
-  VerifyMessageParamsType,
-  VerifyMessageResponseType,
 } from '@shapeshiftoss/metamask-snaps-types'
 import assert from 'assert'
+import { serializeError } from 'eth-rpc-errors'
 
 import { BaseSigner } from '../../common'
 
+const BIP32_HARDENING_CONSTANT = 0x80000000
+
+const ERR_CHAIN_NOT_ADDED_TO_WALLET = 4902 // Internal error, which in the case of wallet_switchEthereumChain call means the chain isn't currently added to the wallet
+
 export abstract class EVMSigner<T extends EVMChainIds> extends BaseSigner<T> {
-  async getAddress({ addressParams }: GetAddressParamsType<T>): Promise<GetAddressResponseType<T>> {
+  async getAddress({
+    addressParams,
+    chainId,
+  }: GetAddressParamsType<T>): Promise<GetAddressResponseType<T>> {
     const { addressNList } = addressParams
+    assert(
+      bip32Like(addressNListToBIP32(addressNList)),
+      'addressParams object does not contain valid BIP32 path',
+    )
     try {
-      const address = await this.signer.ethGetAddress({
-        addressNList,
-        showDisplay: false,
-      } as SignerGetAddressType<T>)
+      // eslint-disable-next-line no-undef
+      const err = await window.ethereum.request({
```

Comment:
<img width="1098" alt="image" src="https://github.com/shapeshift/metamask-snaps/assets/17035424/1aa72c69-c4e2-46d4-aa4e-1b386f24034a">

Reply:
To be fixed in a direct follow-up. We have 165 lint violations in main already.